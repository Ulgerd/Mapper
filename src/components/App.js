import React from 'react';
import Input from './Input.js';
import List from './List.js';
import Mapper from './Mapper.js';
import Header from './Header.js';
import styled from 'styled-components'

const StyledApp = styled.div`
  display: flex;
  justify-content: space-evenly;
  margin-top: 1em;
`

const StyledNav = styled.div`
  width: 20em;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`

export function App() {
  return (
    <div>
      <Header/>
      <StyledApp>
        <StyledNav>
          <Input />
          <List />
        </StyledNav>
        <Mapper />
      </StyledApp>
    </div>
  );
}

export default App;
