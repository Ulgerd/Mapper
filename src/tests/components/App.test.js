import React from 'react';
import { App } from '../../components/App.js';
import renderer from 'react-test-renderer'

jest.mock('../../components/Input', () => 'Input')
jest.mock('../../components/List', () => 'List')
jest.mock('../../components/Mapper', () => 'Mapper')
jest.mock('../../components/Header', () => 'Header')


describe('App component tests',()=>{
    it('renders correctly', () => {
      let cat = renderer
        .create(<App />)
        .toJSON();
      expect(cat).toMatchSnapshot();
    });
});
