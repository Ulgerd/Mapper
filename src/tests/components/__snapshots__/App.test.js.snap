// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests renders correctly 1`] = `
<div>
  <Header />
  <div
    className="sc-bwzfXH jKArlw"
  >
    <div
      className="sc-htpNat envnwQ"
    >
      <Input />
      <List />
    </div>
    <Mapper />
  </div>
  <div
    className="sc-bdVaJa hgsmGs"
  >
    <span>
      Код можно посмотреть тут:
      <a
        href="https://github.com/Ulgerd/Mapper/tree/master/src"
      >
        GitHub Mapper
      </a>
    </span>
    <span>
      Первая часть квалификационного задания:
      <a
        href="https://resume.ulgerd.now.sh/FunBox_Level_1.pdf"
      >
        Level_I
      </a>
    </span>
    <span>
      Резюме:
      <a
        href="https://resume.ulgerd.now.sh/Pylaev_React_Jun.html"
      >
        Resume
      </a>
    </span>
  </div>
</div>
`;
